Android/app/src/androidTest/java/tie/hackathon/travelguide/Tests/DemoTest.java

```diff
@@ -1,29 +1,89 @@
 
+import android.support.test.espresso.Espresso;
+
+import org.junit.Assert;
 import org.junit.Test;
 
-import static android.support.test.espresso.Espresso.onView;
-import static android.support.test.espresso.action.ViewActions.click;
-import static android.support.test.espresso.assertion.ViewAssertions.matches;
-import static android.support.test.espresso.matcher.ViewMatchers.hasSibling;
-import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
-import static android.support.test.espresso.matcher.ViewMatchers.withContentDescription;
-import static android.support.test.espresso.matcher.ViewMatchers.withText;
-import static org.hamcrest.Matchers.allOf;
+import tie.hackathon.travelguide.Constants.Integers;
+import tie.hackathon.travelguide.Constants.Strings;
+import tie.hackathon.travelguide.Constants.Timeouts;
+import tie.hackathon.travelguide.Screens.City;
+import tie.hackathon.travelguide.Screens.LoginPage;
+import tie.hackathon.travelguide.Screens.MainPage;
+import tie.hackathon.travelguide.Screens.MoreCityDetails;
+import tie.hackathon.travelguide.destinations.description.FinalCityInfo;
+import utils.Constants;
 
 /**
  * Created by valentin.boca on 2/21/2018.
  */
 
 public class DemoTest extends EspressoTestBase {
 
-    @Test
-    public void testDemo() throws Exception {
-        onView(withContentDescription("SIGNUP")).perform(click());
+    @Test(timeout = Timeouts.TEST_TIMEOUT_MEDIUM)
+    public void testIfSignUpIsSuccessful() throws Exception {
+        LoginPage.waitForTheLoginPageToBeDisplayed();
+        LoginPage.signUp(Strings.USER, Strings.PHONE_NO, Strings.PASSWORD);
+        LoginPage.tapSignUpButton();
+        MainPage.waitForTheMainScreenToBeDisplayed();
+        MainPage.signOut();
+    }
+
+    @Test(timeout = Timeouts.TEST_TIMEOUT_MEDIUM)
+    public void testIfLogInIsSuccessful() throws Exception {
+        LoginPage.waitForTheLoginPageToBeDisplayed();
+        LoginPage.tapLogInString();
+        LoginPage.logIn(Strings.PHONE_NO, Strings.PASSWORD);
+        LoginPage.tapLogInButton();
+        MainPage.waitForTheMainScreenToBeDisplayed();
+    }
+
+    @Test(timeout = Timeouts.TEST_TIMEOUT_MEDIUM)
+    public void testIfACityFromMainPageIsDisplayed() throws Exception {
+        MainPage.login();
+        MainPage.waitForCityToBeDisplayed(Strings.CITY_MAIN_PAGE, Integers.CITY_ON_RIGHT_MAIN_PAGE);
+        MainPage.clickACityFromMainPage(Strings.CITY_MAIN_PAGE, Integers.CITY_ON_RIGHT_MAIN_PAGE);
+        City.waitForCityInfoToBeDisplayed(Strings.CITY_MAIN_PAGE, Integers.CITY_NAME);
+        Espresso.pressBack();
+        MainPage.signOut();
+    }
+
+    @Test(timeout = Timeouts.TEST_TIMEOUT_MEDIUM)
+    public void testIfFunFactsPageIsDisplayed() throws Exception {
+        MainPage.login();
+        MainPage.waitForCityToBeDisplayed(Strings.CITY_MAIN_PAGE, Integers.CITY_ON_RIGHT_MAIN_PAGE);
+        MainPage.clickACityFromMainPage(Strings.CITY_MAIN_PAGE, Integers.CITY_ON_RIGHT_MAIN_PAGE);
```

Comment:
Pentru codul care se repeta in teste, fa metode in clasa asta, imediat dupa teste.
 Ex: private void loginWaitForAndClickCity() throws Exception {
       MainPage.login();
       MainPage.waitForCityToBeDisplayed(Strings.CITY_MAIN_PAGE, Integers.CITY_ON_RIGHT_MAIN_PAGE);
       MainPage.clickACityFromMainPage(Strings.CITY_MAIN_PAGE, Integers.CITY_ON_RIGHT_MAIN_PAGE);
} 

apoi apelezi metoda asta in toate testele in care ai aceste 3 linii. :)

Reply:
Done.